components/core/routes/Directory.py

```diff
@@ -0,0 +1,274 @@
+        return '{"error":"not authorized"}', 403
+    else:
+        return '{"error":"token error"}', 403
+
+def edit_workspace(name, workspace_id, user_id):
+    token = token_validator(request.headers['token'])
+    if token is not None:
+        try:
+            status = update_workspace(name, workspace_id, user_id)
+            resp = Response(response='{"status":"' + status + '"}', status=200 if status == "success" else 400)
+        except Exception as e:
+            resp = Response('{"error":"' + str(e) + '"}', status=400)
+        resp.headers['token'] = token
+        resp.headers['Access-Control-Expose-Headers'] = 'token'
+        return resp
+    elif token is not None:
+        return '{"error":"not authorized"}', 403
+    else:
+        return '{"error":"token error"}', 403
+
+def remove_workspace(workspace_id, user_id):
+    token = request.headers['token']
+    if token is not None:
+        try:
+            status = delete_workspace(workspace_id, user_id)
+            resp = Response(response='{"status":"' + status + '"}', status=200 if status == "success" else 400)
+        except Exception as e:
+            resp = Response('{"error":"' + str(e) + '"}', status=400)
+        resp.headers['token'] = token
+        resp.headers['Access-Control-Expose-Headers'] = 'token'
+        return resp
+    elif token is not None:
+        return '{"error":"not authorized"}', 403
+    else:
+        return '{"error":"token error"}', 403
+
+
+##################### project endpoints ######################
+
+
+def fetch_projects(workspace_id):
+    token = token_validator(request.headers['token'])
+    if token is not None:
+        try:
+            status = get_projects(workspace_id)
+            if not isinstance(status, str):
+                resp = Response(response=json.dumps(status), status=200)
+            else:
+                resp = Response('{"error":"' + status + '"}', status=400)
+        except Exception as e:
+            resp = Response('{"error":"' + str(e) + '"}', status=400)
+        resp.headers['token'] = token
+        resp.headers['Access-Control-Expose-Headers'] = 'token'
+        return resp
+    elif token is not None:
+        return '{"error":"not authorized"}', 403
+    else:
+        return '{"error":"token error"}', 403
+
+def fetch_project(project_id, workspace_id):
+    token = token_validator(request.headers['token'])
+    if token is not None:
+        try:
+            status = get_project_by_id(project_id, workspace_id)
+            if not isinstance(status, str):
+                resp = Response(response=json.dumps(status), status=200)
+            else:
+                resp = Response('{"error":"' + status + '"}', status=400)
+        except Exception as e:
+            resp = Response('{"error":"' + str(e) + '"}', status=400)
+        resp.headers['token'] = token
+        resp.headers['Access-Control-Expose-Headers'] = 'token'
+        return resp
+    elif token is not None:
```

Comment:
@VibhorCodecianGupta why are we checking this condition again?

Reply:
This is in place for a future use case where I'll put another check in the first if along with the token check which will determine whether the user has correct permissions to carry out the action. So it doesn't make much sense now, but it will once the ACL flow is done